reference-manual/src/main/asciidoc/appendix.adoc

```diff
@@ -283,6 +283,111 @@ The URL used for phone home requests is
 http://phonehome.hazelcast.com/ping
 ----
 
+[appendix]
+= Migration Guides
+
+This appendix provides guidelines for upgrading to a new Hazelcast Jet version.
+See also the link:https://docs.hazelcast.org/docs/jet/release-notes/index.html[Release Notes] document
+for the changes for each Hazelcast Jet release.
+
+== Upgrading to Hazelcast Jet 4.0
+
+Jet 4.0 is a major version release. Major releases allow us to break compatibility in the API, as well as
+to remove previously deprecated parts of the API.
+
+=== Jet on IMDG 4.0
+
+Jet 4.0 uses IMDG 4.0, which is also a major release with its own breaking changes. For details see
+link:https://docs.hazelcast.org/docs/rn/index.html[IMDG Release Notes] and
+link:https://docs.hazelcast.org/docs/latest/manual/html-single/#migration-guides[IMDG Migration Guides].
+
+The most important changes that have affected Jet are as follows:
+
+* Many packages have been renamed, or classes moved from one package to another and these changes have been followed in
+Jet. For details see link:https://docs.hazelcast.org/docs/rn/index.html#4-0[IMDG Release Notes]. The change most obvious
+in Jet is that many classes have been moved from the generic `com.hazelcast.core` package to more specific packages,
+like `com.hazelcast.map` or `com.hazelcast.collection`.
+
+* Serializable variants of `java.util.function` interfaces have been moved to IMDG and removed from Jet: `BiConsumerEx`,
```

Comment:
As I wrote previously I suggest to have notes like `from com.hazelcast.jet.function to com.hazelcast.function` mentioned in this point.